chacra/models/binaries.py

```diff
@@ -52,6 +52,12 @@ def __init__(self, name, project, repo=None, **kw):
             if key in kw.keys():
                 setattr(self, key, kw[key])
         self.repo = repo or self._get_or_create_repo()
+        # ensure that the repo.type is set
+        self._set_repo_type(self.repo)
```

Comment:
see I forgot to add a test for this, and I was thinking on passing it so that I could just test the method directly, but I guess I could still just use self.repo, good point, will fix this and add a test